libraries/AP_Compass/AP_Compass_Backend.cpp

```diff
@@ -7,7 +7,9 @@ extern const AP_HAL::HAL& hal;
 
 AP_Compass_Backend::AP_Compass_Backend(Compass &compass) :
     _compass(compass)
-{}
+{
+    _sem = hal.util->new_semaphore();    
```

Comment:
This should be done in the init() method, not in the constructor because hal may not be constructed yet.